cli/src/pcluster/validators/database_validators.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+from urllib.parse import ParseResult, urlparse
+
+from pcluster.validators.common import FailureLevel, Validator
+
+
+class DatabaseUriValidator(Validator):
+    """Domain address validator."""
+
+    def _validate(self, uri: str):
+        """Validate database URI."""
+        # First, throw error if the URI starts with a "/" (to prevent issues with the
+        # manipulation below
+        if not self._check_trailing_slash(uri):
+            return
+
+        uri_parse = urlparse(uri)
+        if not uri_parse.netloc:
+            # This happens if users provide an URI without explicit scheme followed by ://
+            # (for example 'test.example.com:3306' instead of 'mysql://test.example.com:3306`).
+            uri_parse = urlparse("//" + uri)
+
+        # Throw error if the URI contains a scheme
+        if not self._check_scheme(uri_parse):
+            return
+
+        # Check if netloc can be parsed
+        if not self._check_netloc(uri, uri_parse):
+            return
+
+        # Check if port is provided and acceptable
+        if not self._check_port(uri_parse):
+            return
+
+    def _check_trailing_slash(self, uri: str) -> bool:
```

Comment:
The term **trailing** applies to the end of a string. I believe you meant `check_leading_slash`?

Reply:
You are right. Fixing in the next commit.